### PR TITLE
[SDEM] Many more simplifications

### PR DESCRIPTION
**Pull request by GuillermoCasas**
All tests are running OK

### CODE REVIEW

File: applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py [hunk 1]
```diff
@@ -106,7 +106,11 @@ def SetProjectParameters(self, parameters):
 
         # First, read the parameters generated from the interface
         import swimming_dem_default_input_parameters as only_swimming_defaults
+        import KratosMultiphysics.DEMApplication.dem_default_input_parameters as dem_defaults
```
Comment (philbucher) [inline applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py:109]: :+1:

File: applications/SwimmingDEMApplication/python_scripts/swimming_DEM_PFEM_analysis.py [hunk 2]
```diff
@@ -44,11 +44,11 @@ def TransferWallsFromPfemToDem(self):
     def TransferGravityFromDisperseToFluid(self):
         # setting fluid's body force to the same as DEM's
         if self.project_parameters["custom_fluid"]["body_force_on_fluid_option"].GetBool():
+            gravity = Vector([self.project_parameters["gravity_parameters"]["direction"][i].GetDouble() for i in range(3)])
```
Comment (philbucher) [inline applications/SwimmingDEMApplication/python_scripts/swimming_DEM_PFEM_analysis.py]:
```suggestion
            gravity = self.project_parameters["gravity_parameters"]["direction"].GetVector()
```

File: applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py [hunk 3]
```diff
@@ -181,14 +185,13 @@ def ModifyInputParametersForCoherence(self):
     def TransferBodyForceFromDisperseToFluid(self):
         # setting fluid's body force to the same as DEM's
         if self.project_parameters["custom_fluid"]["body_force_on_fluid_option"].GetBool():
-            body_force = [self.project_parameters["GravityX"].GetDouble(),
-                          self.project_parameters["GravityY"].GetDouble(),
-                          self.project_parameters["GravityZ"].GetDouble()]
-            modulus_of_body_force = math.sqrt(sum(b**2 for b in body_force))
+            gravity = Vector([self.project_parameters["gravity_parameters"]["direction"][i].GetDouble() for i in range(3)])
```
Comment (philbucher) [inline applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py]:
```suggestion
            gravity = self.project_parameters["gravity_parameters"]["direction"].GetVector()
```

File: applications/SwimmingDEMApplication/python_scripts/variables_management.py [hunk 4]
```diff
@@ -79,12 +79,13 @@ def AddExtraProcessInfoVariablesToFluidModelPart(self, parameters, fluid_model_p
...
 
         fluid_model_part.ProcessInfo.SetValue(Kratos.FRACTIONAL_STEP, 1)
-        gravity = Vector(3)
+
         if parameters["custom_fluid"]["body_force_on_fluid_option"].GetBool():
-            gravity[0] = parameters["GravityX"].GetDouble()
-            gravity[1] = parameters["GravityY"].GetDouble()
-            gravity[2] = parameters["GravityZ"].GetDouble()
-        fluid_model_part.ProcessInfo.SetValue(Kratos.GRAVITY, gravity)
+            gravity = Vector([self.project_parameters["gravity_parameters"]["direction"][i].GetDouble() for i in range(3)])
```
Comment (philbucher) [inline applications/SwimmingDEMApplication/python_scripts/variables_management.py]:
```suggestion
            gravity = self.project_parameters["gravity_parameters"]["direction"].GetVector()
```

File: applications/SwimmingDEMApplication/python_scripts/swimming_DEM_PFEM_analysis.py [hunk 5]
```diff
@@ -44,11 +44,11 @@ def TransferWallsFromPfemToDem(self):
...
+            gravity *= self.project_parameters["gravity_parameters"]["modulus"].GetDouble()
 
             for node in self.fluid_model_part.Nodes:
-                node.SetSolutionStepValue(Kratos.VOLUME_ACCELERATION_X, 0, self.project_parameters["GravityX"].GetDouble())
-                node.SetSolutionStepValue(Kratos.VOLUME_ACCELERATION_Y, 0, self.project_parameters["GravityY"].GetDouble())
-                node.SetSolutionStepValue(Kratos.VOLUME_ACCELERATION_Z, 0, self.project_parameters["GravityZ"].GetDouble())
+                node.SetSolutionStepValue(Kratos.VOLUME_ACCELERATION, 0, gravity)
```
Comment (philbucher) [inline applications/SwimmingDEMApplication/python_scripts/swimming_DEM_PFEM_analysis.py]: Better:
~~~py
Kratos.VariableUtils().SetVectorVar(Kratos.VOLUME_ACCELERATION, gravity, self.fluid_model_part.Nodes)
~~~

File: applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py [hunk 6]
```diff
@@ -181,14 +185,13 @@ def ModifyInputParametersForCoherence(self):
...
+            gravity *= self.project_parameters["gravity_parameters"]["modulus"].GetDouble()
+            modulus_of_body_force = math.sqrt(sum(b**2 for b in gravity))
 
             gravity_parameters = self.fluid_parameters['processes']['gravity'][0]['Parameters']
             gravity_parameters['modulus'].SetDouble(modulus_of_body_force)
-            for i, b in enumerate(body_force):
+            for i, b in enumerate(gravity):
```
Comment (philbucher) [inline applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py]: there is also `SetVector`, but I am not sure if it is applicable here

**Review by philbucher — COMMENTED** [review]
some suggestions for improvement

File: applications/SwimmingDEMApplication/python_scripts/variables_management.py (hunk 4 above)
Comment (GuillermoCasas) [reply applications/SwimmingDEMApplication/python_scripts/variables_management.py]: Thanks! I will apply all of these tomorrow

File: applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py [hunk 7]
```diff
@@ -247,7 +250,7 @@ def Initialize(self):
 
             self.swimming_DEM_gid_io = \
             swimming_DEM_gid_output.SwimmingDEMGiDOutput(
-                file_name = self.project_parameters["problem_name"].GetString(),
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py:250]: There is an error here

File: applications/SwimmingDEMApplication/tests/PFEM-DEM_tests/ProjectParameters.json [hunk 8]
```diff
@@ -390,8 +377,7 @@
     "DeltaTimeSafetyFactor"            : 1.0,
     "MaxTimeStep"                      : 2.5e-6,
     "FinalTime"                        : 1.0,
-    "ControlTime"                      : 4.0,
-    "NeighbourSearchFrequency"         : 1,
+        "NeighbourSearchFrequency"         : 1,
     "TestType"                         : "None",
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/tests/PFEM-DEM_tests/ProjectParameters.json]: Indent

File: applications/SwimmingDEMApplication/test_examples/PFEMcoupledDEMexample.gid/ProjectParameters.json [hunk 9]
```diff
@@ -333,7 +333,7 @@
             "restart_file_name"   : "PFEMcoupledDEMexample",
             "restart_file_label"  : "step",
             "output_control_type" : "step",
-            "output_frequency"    : 1,
+            "output_interval"    : 1,
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/test_examples/PFEMcoupledDEMexample.gid/ProjectParameters.json]: Indent

**Comment by GuillermoCasas** [comment]
I have mistakenly included the promised changes in another branch. But it is ok, they are all aesthetic.

**Review by GuillermoCasas — COMMENTED** [review]
> I have mistakenly included the promised changes in another branch. But it is ok, they are all aesthetic.

Ok, I just cherrypìcked the changes from the other branch

**Review by farrufat-cimne — APPROVED** [review]

**Pull request merged**